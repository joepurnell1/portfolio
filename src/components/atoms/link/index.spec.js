import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Link from './index';

const theme = {
  Link: {
    accent: '#BADA55',
  },
  Global: {
    background: '#1A1A1A',
  },
};

describe('link component', () => {
  it('should render default as expected', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Link href="LinkUrl">We will, we will</Link>
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render inverted as expected', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Link href="LinkUrl" invert>Link you, Link you</Link>
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
