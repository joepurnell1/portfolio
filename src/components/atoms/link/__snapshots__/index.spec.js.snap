// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link component should render default as expected 1`] = `
<a
  className="sc-bdVaJa cVqcJE"
  href="LinkUrl"
>
  We will, we will
</a>
`;

exports[`link component should render inverted as expected 1`] = `
<a
  className="sc-bdVaJa JsCiN"
  href="LinkUrl"
>
  Link you, Link you
</a>
`;
